src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -108,15 +110,18 @@ public AzureAppConfigurationProvider(IConfigurationClientManager configClientMan
             _options = options ?? throw new ArgumentNullException(nameof(options));
             _optional = optional;
 
-            IEnumerable<KeyValueWatcher> watchers = options.ChangeWatchers.Union(options.MultiKeyWatchers);
+            IEnumerable<KeyValueWatcher> watchers = options.IndividualKvWatchers.Union(options.FeatureFlagWatchers);
 
-            if (watchers.Any())
+            bool hasWatchers = watchers.Any();
+            TimeSpan minWatcherRefreshInterval = hasWatchers ? watchers.Min(w => w.RefreshInterval) : TimeSpan.MaxValue;
+
+            if (options.RegisterAllEnabled)
```

Comment:
The embedded ternary condition in the `else` statement makes more sense to be a part of the upper if/else.

```
if (registerAllEnabled)
{
    
}
else if (watchers.Any())
{

}
else
{
    
}
```